src/c++/perf_analyzer/test_idle_timer.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "doctest.h"
+#include "idle_timer.h"
+
+namespace triton { namespace perfanalyzer {
+
+TEST_CASE("idle_timer: basic usage")
+{
+  IdleTimer timer;
+  CHECK(0 == timer.GetIdleTime());
```

Comment:
Can you flip the operands? It's more English-readable the other way, IMO.

Reply:
This way avoids accidental assignment and I think is the better way to handle this.

Reply:
fixed

Reply:
doctest actually doesn't allow assignments inside the `CHECK()` macro, so we're safe 👍:

```cpp
// a.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

TEST_CASE("test") {
  int x{5};
  CHECK(x = 5);
}
```
```bash
$ g++ a.cpp
In file included from a.cpp:3:
doctest.h: In instantiation of 'doctest::detail::Expression_lhs<L>& doctest::detail::Expression_lhs<L>::operator=(const R&) [with R = int; L = int&]':
a.cpp:7:3:   required from here
doctest.h:1348:42: error: static assertion failed: Expression Too Complex Please Rewrite As Binary Comparison!
 1348 |         static_assert(deferred_false<R>::value,                                                    \
      |                                          ^~~~~
doctest.h:1499:9: note: in expansion of macro 'DOCTEST_FORBIT_EXPRESSION'
 1499 |         DOCTEST_FORBIT_EXPRESSION(Expression_lhs, =)
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~
```